### PR TITLE
[caffe2][python] Expose tensorshape to python

### PR DESCRIPTION
**Pull request by bwasti**
Exposes the TensorShape information to the python API

### CODE REVIEW

File: caffe2/python/pybind_state.cc [hunk 1]
```diff
@@ -1107,6 +1107,27 @@ void addGlobalMethods(py::module& m) {
...
       });
+  m.def(
+      "get_operator_input_blob_shape",
+      [](const py::bytes& op_def, const std::vector<string>& input_blobs) {
+        CAFFE_ENFORCE(gWorkspace);
+        OperatorDef def;
+        CAFFE_ENFORCE(
+            ParseProtoFromLargeString(op_def.cast<std::string>(), &def),
+            "Couldn't parse operator proto.");
+        const auto op_type = def.type();
```
Comment (jamesr66a) [inline caffe2/python/pybind_state.cc]: Can you just pass def.type() as the argument to Schema() below?

File: caffe2/python/pybind_state.cc [hunk 2]
```diff
@@ -1107,6 +1107,26 @@ void addGlobalMethods(py::module& m) {
...
+  m.def(
+      "get_operator_input_blob_shape",
+      [](const py::bytes& op_def, const std::vector<string>& input_blobs) {
+        CAFFE_ENFORCE(gWorkspace);
+        OperatorDef def;
+        CAFFE_ENFORCE(
+            ParseProtoFromLargeString(op_def.cast<std::string>(), &def),
+            "Couldn't parse operator proto.");
+        auto* schema = OpSchemaRegistry::Schema(def.type());
+        CAFFE_ENFORCE(schema);
```
Comment (bddppq) [inline caffe2/python/pybind_state.cc]: schema doesn't seem to be needed? (and thus op_def)

File: caffe2/python/pybind_state.cc [hunk 3]
```diff
@@ -1107,6 +1107,26 @@ void addGlobalMethods(py::module& m) {
...
+        std::unordered_map<string, std::vector<int64_t>> shapes;
+        for (const auto& blob_name : input_blobs) {
+          auto* blob = gWorkspace->GetBlob(blob_name);
+          auto ts = GetTensorShapeOfBlob(blob);
+          for (auto dim : ts.dims()) {
```
Comment (bddppq) [inline caffe2/python/pybind_state.cc]: `shapes[blob_name] = ts.dims()`

File: caffe2/python/pybind_state.cc [hunk 4]
```diff
@@ -1107,6 +1107,26 @@ void addGlobalMethods(py::module& m) {
...
+        CAFFE_ENFORCE(gWorkspace);
+        OperatorDef def;
+        CAFFE_ENFORCE(
+            ParseProtoFromLargeString(op_def.cast<std::string>(), &def),
+            "Couldn't parse operator proto.");
+        auto* schema = OpSchemaRegistry::Schema(def.type());
+        CAFFE_ENFORCE(schema);
+        std::unordered_map<string, std::vector<int64_t>> shapes;
+        for (const auto& blob_name : input_blobs) {
+          auto* blob = gWorkspace->GetBlob(blob_name);
```
Comment (bddppq) [inline caffe2/python/pybind_state.cc]: need to check blob exists

File: caffe2/python/pybind_state.cc [hunk 5]
```diff
@@ -1107,6 +1107,26 @@ void addGlobalMethods(py::module& m) {
...
+          auto ts = GetTensorShapeOfBlob(blob);
```
Comment (bddppq) [inline caffe2/python/pybind_state.cc]: `GetTensorShapeOfBlob` constructs a TensorShape struct. Shouldn't we be able to directly to get the shape of the Tensor in c++? (of course we need to check whether the blob is of type Tensor first).

**Pull request closed without merging**